server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQClusterTest.java

```diff
@@ -0,0 +1,286 @@
+
+import com.github.fge.lambdas.Throwing;
+import com.jayway.awaitility.Awaitility;
+import com.jayway.awaitility.Duration;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+
+@ExtendWith(DockerClusterRabbitMQExtention.class)
+class RabbitMQClusterTest {
+
+    private static final String QUEUE = "queue";
+
+    @Nested
+    class ClusterSharing {
+
+        private ConnectionFactory node1ConnectionFactory;
+        private ConnectionFactory node2ConnectionFactory;
+        private Connection node1Connection;
+        private Connection node2Connection;
+        private Channel node1Channel;
+        private Channel node2Channel;
+
+        @BeforeEach
+        void setup(DockerRabbitMQCluster cluster) throws IOException, TimeoutException {
+            node1ConnectionFactory = cluster.getRabbitMQ1().connectionFactory();
+            node2ConnectionFactory = cluster.getRabbitMQ2().connectionFactory();
+            node1Connection = node1ConnectionFactory.newConnection();
+            node2Connection = node2ConnectionFactory.newConnection();
+            node1Channel = node1Connection.createChannel();
+            node2Channel = node2Connection.createChannel();
+        }
+
+        @AfterEach
+        void tearDown() {
+            closeQuietly(node1Channel, node2Channel, node1Connection, node2Connection);
+        }
+
+        @Test
+        void rabbitMQManagerShouldReturnThreeNodesWhenAskingForStatus(DockerRabbitMQCluster cluster) throws Exception {
+            String stdout = cluster.getRabbitMQ1().container()
+                .execInContainer("rabbitmqctl", "cluster_status")
+                .getStdout();
+
+            assertThat(stdout)
+                .contains(
+                    DockerClusterRabbitMQExtention.RABBIT_1,
+                    DockerClusterRabbitMQExtention.RABBIT_2,
+                    DockerClusterRabbitMQExtention.RABBIT_3);
+        }
+
+        @Test
+        void queuesShouldBeShared(DockerRabbitMQCluster cluster) throws Exception {
+            node1Channel.exchangeDeclare(EXCHANGE_NAME, DIRECT, DURABLE);
+            node1Channel.queueDeclare(QUEUE, DURABLE, !EXCLUSIVE, !AUTO_DELETE, ImmutableMap.of()).getQueue();
+            node1Channel.queueBind(QUEUE, EXCHANGE_NAME, ROUTING_KEY);
+
+            int nbMessages = 10;
+            IntStream.range(0, nbMessages)
+                .mapToObj(i -> asBytes(String.valueOf(i)))
+                .forEach(Throwing.consumer(
+                    bytes -> node1Channel.basicPublish(EXCHANGE_NAME, ROUTING_KEY, NO_PROPERTIES, bytes)));
+
+
+            InMemoryConsumer consumer2 = new InMemoryConsumer(node2Channel);
+            node2Channel.basicConsume(QUEUE, consumer2);
+
+            awaitAtMostOneMinute.until(() -> consumer2.getConsumedMessages().size() == nbMessages);
+
+            List<Integer> expectedResult = IntStream.range(0, nbMessages).boxed().collect(Collectors.toList());
```

Comment:
Guavate.toImmutableList()?

Reply:
How would you explain the need for an immutable list in such a tiny scope ?

Reply:
Because we should not bother with such question of scope and make the raquirement for immutable collections the default.

IMO I should not have to justify using it, but you should in order to use mutable data. (immutable >> mutable, right?)

Reply:
Yes, we can apply the rule blindly, it doesn't hurt much.